src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/SQLCompiler.txt

```diff
@@ -0,0 +1,76 @@
+// These tests originated in the SQL compiler test suite.  
+// These all pass with C# too, so moved here instead for coverage for all backends and to ensure consistency.
+
+// IsError inside If
+>> If(true, IsError(1/0 + 1))
+true
+
+>> If(IsError(1/0), 1, 2)
+1
+
+>> Switch(IsError(1/0), true, 1, 2)
+1
+
+// If inside IsError
+>> IsError(If(false, 1/0, 1/0))
+true 
+
+>> IsError(If(true, 1/0, 1/0))
+true
+
+>> IsError(Switch(1, 1, 1/0, 1/0))
+true
+
+// IsError inside If inside IsError
+>> IsError(If(false, IsError(1/0), IsError(1/0)))
+false
+
+>> IsError(1/2 + 1) && IsError(1/0)
+false
+
+>> IsError(1/2 + 2)
+false
+
+>> IsError(1/0 + 1)
+true
+
+>> IfError(If(true, 1/0, 1/0), 2)
+2
+
+>> IsError(If(IfError(1/0 = 1, true), 1/0, 1/0))
+true
+
+// check string literals are sanitized
+>> "Robert '; DROP TABLE Students; --"
+"Robert '; DROP TABLE Students; --"
+
+// check SQL wildcards are handled
+>> "%_" in "SQL wildcards"
+false
+
+// ensure Substitute is case sensitive
+>> Substitute("HelloHELLOhello", "he", "je")
+"HelloHELLOjello"
+
+>> Substitute("HelloHelloHello", "he", "je", 2)
+"HelloHelloHello"
+
+>> Substitute("HelloHELLOhelloHelloHELLOhello", "he", "je", 2)
+"HelloHELLOhelloHelloHELLOjello"
+
+>> Value( "1" & "2" & "3" ) * Value( "4" & "5" & "6" )
+56088
+
+>> 123 * 456
+56088
+
+>> Value( "1" & Text( Value( "3" ), "####" ) ) * -2
+-26
+
+>> Value( "1" & Text( Value( Mid( ".3.", 2, 1) ), "####" ) ) * -2
+-26
+
+>> Value( "1" & Text( Value( Upper( Mid( ".3.", 2, 1) ) ), "####" ) ) * -2
+-26
+
+// Test: 4,000 character string field concat together
```

Comment:
Where is the test? Only the comment is here

Reply:
Removed.  The test needs to be against an actual Dataverse instance.